README.md

```diff
@@ -616,13 +620,33 @@ ISB rotates the signing key once a week in production. To help testing, the sign
 ![Key rotation](./key_rotation.png?raw=true)
 
 - CREATE, new key is created
-- WARM-UP, key is published in the JWKS but not in use
+- WARM-UP, key is published in the signed JWKS but not in use
 - NORMAL USE, key is in use
-- COOLDOWN, key is still published in the JWKS and is not in use
-- TAIL, key is not published anymore in the JWKS, key is not in use
+- COOLDOWN, key is still published in the signed JWKS and is not in use
+- TAIL, key is not published anymore in the signed JWKS, key is not in use
 - DELETION, key is deleted
 
-Caching the keys fetched from the JWKS endpoint is a good idea, but make sure that the refresh mechanism supports the ISB lifecycle, and there is a forced cache refresh in case key is not found.
+Caching the keys fetched from the signed JWKS endpoint is a good idea, but make sure that the refresh mechanism supports the ISB lifecycle, and there is a forced cache refresh in case key is not found in the cache.
+
+About SP key rotation.
+
+SP should rotate it's OIDC signing and encryption keys regularly. The rotation of the signing key should follow same principle as described above for the ISB.
+
+The SP's encyption key rotation should follow this principle
+
+![encyption key rotation](./encryption_key_rotation.png?raw=true)
+
+- CREATE, new key is created
+- WARM-UP, key is not published and not in use
```

Comment:
Pitäiskö warm-up olla näin: "key is not published but it is deployed side-by-side with the old key so that it can be used for decrypting immediately after it is published"?